### PR TITLE
Expose `collect` method in `Loader`

### PR DESCRIPTION
**Pull request by hisener**
This PR target https://github.com/PicnicSupermarket/jolo/pull/9.

### CODE REVIEW

File: src/main/java/tech/picnic/jolo/Loader.java [hunk 1]
```diff
@@ -65,6 +66,11 @@ public void next(Record record) {
     return get().stream();
   }
 
+  /** Returns all objects loaded by this loader. */
```
Comment (Stephan202) [inline src/main/java/tech/picnic/jolo/Loader.java]:
```suggestion
  /** Collects and then returns all objects loaded by this loader. */
```
The `Collector` may do anything, so I think that's worth calling out.

**Review by Mordavolt — APPROVED** [review]
Great, thanks!

**Review by Stephan202 — APPROVED** [review]

**Comment by hisener** [comment]
Rebased.

**Comment by Stephan202** [comment]
Suggested commit message:
```
Introduce `Loader#collect(Collector<? super T, A, R> collector)` (#11)
```

**Comment by Stephan202** [comment]
Rebased; PR now targets `master`.

**Pull request merged**